src/components/JdsDataTable/DataTable.vue

```diff
@@ -139,18 +166,41 @@ export default {
       type: Boolean,
       default: false
     },
+
+    /**
+     * Allow DataTable to do row selection
+     */
+    showSelect: {
+      type: Boolean,
+      default: false
+    },
+
+    /**
+     * Define what key or property used
+     * when item is being selected.
+     */
+    itemKey: {
+      type: String,
+      default: 'id'
+    }
   },
+
   computed: {
     isDataEmpty() {
       if (Array.isArray(this.items) && this.items.length === 0) {
         return true
       }
       return false
     }
-  }
+  },
+
+  mounted() {
+    this.columnLength = this.$refs.tableHeader.childElementCount
```

Comment:
Is `columnLength` expected to be not reactive?.

Reply:
I am honestly not so sure about this one, is there a case that user would dynamically change row column? @adrianpdm 

if so, I think I should make `columnLength` reactive

Reply:
after some consideration I think it's better to add computed property for this (in case table column change for some reason) @adrianpdm 
fixed [here](https://github.com/jabardigitalservice/jds-design-system-vue-lib/pull/90/commits/72aa85e4220a5c88a4389eae7546b44c7fe5ea23)

Reply:
my concern are:
- column length is called under `mounted` hook, which is called **only once** during a component lifecycle.
- $refs are not reactive either

have you ever found that toggling between element that is not keyed, didn't actually re-create the component, but instead only update its props?.

Reply:
Yes you're right. I didn't consider the possibility that the column might change, so I thought it would be fine to count the child elements inside `<tr>` once (that's why I used the **mounted** hook in that case)

thanks for the insight @adrianpdm 👍🏼